src/dbnode/storage/shard.go

```diff
@@ -2725,77 +2727,92 @@ func (s *dbShard) AggregateTiles(
 	}
 
 	nextVolume := latestTargetVolume + 1
-	writer, err := fs.NewStreamingWriter(
-		fs.StreamingWriterOptions{
-			NamespaceID:         s.namespace.ID(),
-			ShardID:             s.ID(),
-			Options:             s.opts.CommitLogOptions().FilesystemOptions(),
-			BlockStart:          opts.Start,
-			BlockSize:           s.namespace.Options().RetentionOptions().BlockSize(),
-			VolumeIndex:         nextVolume,
-			PlannedRecordsCount: uint(maxEntries),
-		},
-	)
-	if err != nil {
-		return 0, err
-	}
-
-	if err := writer.Open(); err != nil {
+	writerOpenOpts := fs.StreamingWriterOpenOptions{
+		NamespaceID:         s.namespace.ID(),
+		ShardID:             s.ID(),
+		BlockStart:          opts.Start,
+		BlockSize:           s.namespace.Options().RetentionOptions().BlockSize(),
+		VolumeIndex:         nextVolume,
+		PlannedRecordsCount: uint(maxEntries),
+	}
+	if err = writer.Open(writerOpenOpts); err != nil {
 		return 0, err
 	}
 
 	var (
 		processedTileCount atomic.Int64
 		multiErr           xerrors.MultiError
-		dp                 ts.Datapoint
+		annotationPayload  annotation.Payload
+		downsampledValues  = make([]tile.DownsampledValue, 0, 4)
```

Comment:
nit: comment for `4`, or better yet have it as a commented constant, e.g. `tile.MaxDownsampledValues`

Reply:
👍